platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ResolvedFilesApiIntegrationTest.groovy

```diff
@@ -20,6 +20,26 @@ import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
 import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
 
 class ResolvedFilesApiIntegrationTest extends AbstractHttpDependencyResolutionTest {
+
+    // These methods return Set<File>, and do not carry task dependencies
+    private static final List<String> FILE_EXPRESSION_LIST = [
+        "configurations.compile.files",
+        "configurations.compile.resolve()",
+    ]
+
+    // These methods return FileCollection, and are expected to track task dependencies
+    private static final List<String> FILE_COLLECTION_EXPRESSION_LIST = [
+        "configurations.compile",
+        "configurations.compile.incoming.files",
+        "configurations.compile.incoming.artifacts.artifactFiles",
+        "configurations.compile.incoming.artifactView { }.files",
+        "configurations.compile.incoming.artifactView { }.artifacts.artifactFiles",
+        "configurations.compile.incoming.artifactView { componentFilter { true } }.files",
+        "configurations.compile.incoming.artifactView { componentFilter { true } }.artifacts.artifactFiles",
+    ]
```

Comment:
🤔 Is this distinction noted in the javadoc for these methods?  

Also, I think this distinction is work a Best Practice/Footgun.

Reply:
Some tests verify task dependencies, and can only execute against the second list, while others dont and execute against all elements